graphql/database/widgets/bookmarkWidget.js

```diff
@@ -0,0 +1,183 @@
+import database from '../database';
+import array from 'lodash/array';
+
+import { 
+  getUserWidget,
+  updateUserWidgetData 
+} from './widgets';
+
+import Async from 'asyncawait/async';
+import Await from 'asyncawait/await';
+
+import { logger } from '../../utils/dev-tools';
+
+/*
+ * Helper Class to manage the BookmarkWidget.
+ */
+class BookmarkWidget {
+
+  /**
+   * Creates a BookmarkWidget instance from the userId and the 
+   * widgetId.
+   * @param {string} userId - The user id.
+   * @param {string} widgetId - The widget id.
+   */
+  constructor(userId, widgetId, widget) {
+    this.widget = widget;
+    this.userId = userId;
+    this.widgetId = widgetId;
+    this.bookmarks = widget.data.bookmarks;
+
+    /**
+    * Updates the bookmarks of the current UserWidget.
+    * @return {Object<UserWidget>}  Returns the UserWidget with 
```

Comment:
@rchall-gladly I believe the `@return` should be `Promise<>` rather than `Object<>` for this method and all others that call `Async`?